velox/common/encode/Base64.h

```diff
@@ -32,125 +29,125 @@ class Base64 {
+  static void decode(const char* input, size_t inputSize, char* outputBuffer);
+  /// Decodes the specified number of characters from the 'input' and writes the
+  /// result to the 'outputBuffer'.
+  static size_t decode(
+      const char* input,
+      size_t inputSize,
+      char* outputBuffer,
+      size_t outputSize);
+
+  /// Decodes the input Base64 URL encoded string.
+  static std::string decodeUrl(folly::StringPiece encodedText);
   /// Decodes the specified URL encoded payload and writes the result to the
   /// 'output'.
   static void decodeUrl(
       const std::pair<const char*, int32_t>& payload,
       std::string& output);
+  /// Decodes the specified number of characters from the 'input' using URL
+  /// encoding and writes the result to the 'outputBuffer'
+  static void decodeUrl(
+      const char* input,
+      size_t inputSize,
+      char* outputBuffer,
+      size_t outputSize);
 
-  /// Decodes the specified URL encoded text.
-  static std::string decodeUrl(folly::StringPiece text);
-
-  /// Decodes the specified number of characters from the 'src' and writes the
-  /// result to the 'dst'.
-  static size_t
-  decode(const char* src, size_t src_len, char* dst, size_t dst_len);
+  /// Calculates the encoded size based on input size.
+  static size_t calculateEncodedSize(size_t inputSize, bool withPadding = true);
 
-  /// Decodes the specified number of characters from the 'src' using URL
-  /// encoding and writes the result to the 'dst'.
-  static void
-  decodeUrl(const char* src, size_t src_len, char* dst, size_t dst_len);
+  /// Calculates the decoded size based on encoded input and adjusts the input
+  /// size for padding.
+  static size_t calculateDecodedSize(const char* input, size_t& inputSize);
 
  private:
-  /// Checks if there is padding in encoded data.
-  static inline bool isPadded(const char* data, size_t len) {
-    return (len > 0 && data[len - 1] == kPadding);
+  // Padding character used in encoding.
+  static const char kPadding = '=';
+
+  // Checks if the input Base64 string is padded.
+  static inline bool isPadded(const char* input, size_t inputSize) {
+    return (inputSize > 0 && input[inputSize - 1] == kPadding);
   }
 
-  /// Counts the number of padding characters in encoded data.
-  static inline size_t numPadding(const char* src, size_t len) {
+  // Counts the number of padding characters in encoded input.
+  static inline size_t numPadding(const char* input, size_t inputSize) {
     size_t numPadding{0};
-    while (len > 0 && src[len - 1] == kPadding) {
+    while (inputSize > 0 && input[inputSize - 1] == kPadding) {
       numPadding++;
-      len--;
+      inputSize--;
     }
     return numPadding;
   }
 
-  /// Performs a reverse lookup in the reverse index to retrieve the original
-  /// index of a character in the base.
-  static uint8_t base64ReverseLookup(char p, const ReverseIndex& reverseIndex);
+  // Reverse lookup helper function to get the original index of a Base64
```

Comment:
For multi-line comments in headers, we need `///`. 
https://github.com/facebookincubator/velox/blob/main/CODING_STYLE.md

Reply:
I missed the private scope. This is fine. Only functions that need documentation need `///`